### PR TITLE
feat(sequencer-utils): generate example genesis state

### PR DESCRIPTION
**Pull request by SuperFluffy**
## Summary
Provides a `generate-genesis-state` subcommand to sequencer-utils.

## Background
We had a `test-genesis.json` file committed to the repo that was not checked to be in-sync with the most recent version of sequencer's actual `GenesisState`. The subcommand `generate-genesis-state` removes the need to keep the file in sync because one can always generate one that matches the current format.

## Changes
- remove `astria-sequencer/test-genesis-app-state.json`
- move sequencer's `GenesisState` to the new `astria-core::sequencer` submodule. Because this config is effectively a public API, it is justified to be outside of the `astria-sequencer` binary crate.

## Testing
It is a very thin binary which can be run by hand. The specific format of `GenesisState` not changing is ensured by a snapshot test that was moved from `astria-sequencer` to `astria-core`.

## Related Issues
Closes https://github.com/astriaorg/astria/issues/1212

### CODE REVIEW

File: crates/astria-sequencer/README.md [hunk 1]
```diff
@@ -101,10 +101,14 @@ abci-cli query --path=accounts/nonce/<ADDRESS> 0x00
 # initialize the node
 cometbft init
 
-# inside astria-sequencer, update the genesis file to include genesis
-# application state
-../../target/debug/astria-sequencer-utils \
-    --genesis-app-state-file=test-genesis-app-state.json \
+# generate a sequencer genesis file
+cargo run -p astria-sequencer-utils -- generate-genesis-state -o genesis-state-json
```
Comment (noot) [inline crates/astria-sequencer/README.md:105]:
```suggestion
cargo run -p astria-sequencer-utils -- generate-genesis-state -o genesis-state.json
```

File: crates/astria-sequencer/README.md [hunk 2]
```diff
@@ -101,10 +101,14 @@ abci-cli query --path=accounts/nonce/<ADDRESS> 0x00
...
+
+# Edit the generated `genesis-state.json` as desired
+
+# ipdate the cometbft genesis file to include the sequencer genesis application state
```
Comment (noot) [inline crates/astria-sequencer/README.md:109]:
```suggestion
# update the cometbft genesis file to include the sequencer genesis application state
```

File: crates/astria-sequencer/src/lib.rs [hunk 3]
```diff
@@ -9,7 +9,7 @@ mod build_info;
 pub(crate) mod component;
 pub mod config;
 pub(crate) mod fee_asset_change;
-pub(crate) mod genesis;
+// pub(crate) mod genesis;
```
Comment (noot) [inline crates/astria-sequencer/src/lib.rs:12]: remove

**Review by noot — COMMENTED** [review]
can you update the astria-sequencer justfile `run-cometbft` command as well? otherwise, looks good!

File: crates/astria-sequencer/src/lib.rs (hunk 3 above)
Comment (SuperFluffy) [reply crates/astria-sequencer/src/lib.rs]: Done in  0dfad0b2

**Review by noot — APPROVED** [review]

**Pull request merged**